Upgrade EUI to v95.1.0

cee-chen opened the pull request:
`v95.0.0-backport.0` ⏩ `v95.1.0-backport.0`

This PR primarily concerns converting multiple common/building block form control components to Emotion (text, number, and search fields). This means that custom CSS or direct `className` usage of these form controls **should be manually QA'd** to ensure they still look the same before visually, with no regressions.

_[Questions? Please see our Kibana upgrade FAQ.](https://github.com/elastic/eui/blob/main/wiki/eui-team-processes/upgrading-kibana.md#faq-for-kibana-teams)_ 

---

## [`v95.1.0`](https://github.com/elastic/eui/releases/v95.1.0)

- Updated `EuiFormControlLayout` to automatically pass icon padding affordance down to child `input`s ([#7799](https://github.com/elastic/eui/pull/7799))

**Bug fixes**

- Fixed broken focus/invalid styling on compressed `EuiDatePickerRange`s ([#7770](https://github.com/elastic/eui/pull/7770))

**CSS-in-JS conversions**

- Converted `EuiFieldText` to Emotion ([#7770](https://github.com/elastic/eui/pull/7770))
- Updated the autofill colors of Chrome (and other webkit browsers) to better match EUI's light and dark mode ([#7776](https://github.com/elastic/eui/pull/7776))
- Converted `EuiFieldNumber` to Emotion ([#7802](https://github.com/elastic/eui/pull/7802))
- Converted `EuiFieldSearch` to Emotion ([#7802](https://github.com/elastic/eui/pull/7802))
- Converted `EuiFieldPassword` to Emotion ([#7802](https://github.com/elastic/eui/pull/7802))
- Converted `EuiTextArea` to Emotion ([#7812](https://github.com/elastic/eui/pull/7812))
- Converted `EuiSelect` to Emotion ([#7812](https://github.com/elastic/eui/pull/7812))
- Converted `EuiSuperSelect` to Emotion ([#7812](https://github.com/elastic/eui/pull/7812))

## [`v95.1.0-backport.0`](https://github.com/elastic/eui/releases/v95.1.0-backport.0)

**This is a backport release only intended for use by Kibana.**

- Updated `EuiSteps` to support a new `titleSize="xxs"` style, which outputs the same title font size but smaller unnumbered step indicators ([#7813](https://github.com/elastic/eui/pull/7813))
- Updated `EuiStepsHorizontal` to support a new `size="xs"` style, which outputs smaller unnumbered step indicators ([#7813](https://github.com/elastic/eui/pull/7813))
- Updated `EuiStepNumber` to support new `titleSize="none"` which omits rendering step numbers, and will only render icons ([#7813](https://github.com/elastic/eui/pull/7813))

cee-chen committed:
[EuiFieldText] Remove direct className usage, replace with component

cee-chen committed:
[EuiFieldText] Remove custom CSS

- in favor of icon configuration which sets the icon on the desired side (right) and automatically accounts for other icons/padding

- remove `[readonly]` CSS, as the component isn't even setting a readonly prop so I'm not sure what that CSS is doing??

cee-chen committed:
[EuiFieldText] Remove deprecated className from test selector

cee-chen committed:
[EuiSuperSelect] Replace direct className usage

- no longer has any styles attached as of Emotion conversion, so manually apply styles instead

cee-chen committed:
Upgrade EUI to v95.1.0

cee-chen commented:
/ci

cee-chen committed:
[EuiSuperSelect] Fix option type imports

- should be using the top level import, not reaching directly into the file

cee-chen committed:
Update snapshots

cee-chen committed:
Fix/update Enzyme tests affected by new Emotion wrappers

cee-chen commented:
/ci

cee-chen committed:
Update Enzyme snapshot

- prefer `.render()`ed snapshot instead Enzyme wrapper snapshot

cee-chen commented:
/ci

cee-chen commented:
/ci

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

dhurley14 reviewed (approved):
detection engine team changes LGTM. Thanks. Just snapshot updates.

jennypavlova reviewed (commented):
The code changes look good! 💯  
I am not 100% sure if it's related to the PR but I saw this while testing:
- The left paddings of the control drop-downs are missing (Infra -> Hosts)
![image](https://github.com/elastic/kibana/assets/14139027/c134dbfe-b574-4a30-a320-00b8a97c7194)

- Some search bars have borders on click some do not (APM & Infra -> Alerts)

https://github.com/elastic/kibana/assets/14139027/0c88ab99-e67c-4902-8e2b-b501314af381


https://github.com/elastic/kibana/assets/14139027/335379d4-fc1c-4649-b7ab-3076ed460b32

Heenawter commented:
> The left paddings of the control drop-downs are missing (Infra -> Hosts)

FYI - I looked into this, and this is **most likely** a result of https://github.com/elastic/kibana/pull/184533 + how the Hosts page modifies the `renderPrepend` in `x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/components/search_bar/controls_content.tsx` :) Don't think it's related to this EUI update.

jennypavlova commented:
> FYI - I looked into this, and this is most likely a result of https://github.com/elastic/kibana/pull/184533 + how the Hosts page modifies the renderPrepend in .../pages/metrics/hosts/components/search_bar/controls_content.tsx :) Don't think it's related to this EUI update.

@Heenawter Thanks for pointing that out! Looks like it's a recent bug as I couldn't see it on an earlier main version. Do we have an open issue for fixing that? 

@cee-chen Sorry for that, I wasn't sure where this bug came from and I saw it on the branch, I will approve 👍

cee-chen committed:
Fix legacy theme CSS overridding Emotion focus border/box-shadow CSS

From a scan of this theme, this likely shouldn't be affecting _all_ `input` elements wholesale in any case

Heenawter reviewed (approved):
Canvas snapshot changes + quick review of Dashboard to ensure no style impacts. LGTM :+1:

cee-chen commented:
@jennypavlova Thanks so much for the thorough QA Jenny and Hannah, y'all rock! The missing border on focus issue should be fixed now with the latest commit, and I appreciate you flagging that so much 🙇 

Feel free to ping us if we can help out with the odd padding behavior of form prepend/append nodes. As Hannah noted, those should not have been affected by this PR (not yet migrated to Emotion), although that will be happening soon.

jennypavlova commented:
@cee-chen Thank you for fixing the borders, they look fine now! 💯

cee-chen committed:
Merge branch 'main' into eui/v95.1.0

saarikabhasi reviewed (approved):
Search changes LGTM. Tested locally.
